src/dialogs/mqtt-device-debug-info-dialog/mqtt-messages.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  LitElement,
+  html,
+  TemplateResult,
+  customElement,
+  property,
+} from "lit-element";
+import { safeDump } from "js-yaml";
+import { MQTTMessage } from "../../data/mqtt";
+
+@customElement("mqtt-messages")
+class MQTTMessages extends LitElement {
+  @property() public messages!: MQTTMessage[];
+  @property() public showAsYaml: boolean = false;
+  @property() public showDeserialized: boolean = false;
+  @property() public subscribedTopic!: string;
+  @property() public summary!: string;
+  @property() private _open: boolean = false;
+  @property() private _payloadsJson!: Array<object | undefined>;
+  @property() private _showTopic: boolean = false;
+
+  protected firstUpdated(): void {
+    const nPayloads = this.messages.length;
+    this._payloadsJson = new Array(nPayloads);
+    for (let i = 0; i < nPayloads; i++) {
+      const payload = this.messages[i].payload;
+      const topic = this.messages[i].topic;
+
+      // If any message's topic differs from the subscribed topic, show topics + payload
+      if (this.subscribedTopic !== topic) {
+        this._showTopic = true;
+      }
+
+      let o = payload;
+
+      // If the payload is a string, determine if the payload is valid JSON and if it
+      // is, assign the object representation to this._payloadJson.
+      if (typeof payload === "string") {
+        try {
+          o = JSON.parse(payload);
```

Comment:
Can we move the `json` parsing to `_renderSinglePayload`?

Reply:
`JSON.parse(this.messages[i].payload)` will give the same as `this._payloadsJson[i]` right?

Reply:
Refactored to not parse until the payload is rendered